gromacs/fileformats/blocks.py

```diff
@@ -0,0 +1,461 @@
+                self.gromacs['param']['kub'] = self.charmm['param']['kub'] * 2 * 4.184 * 10 * 10
+                self.gromacs['param']['s0'] = self.charmm['param']['s0'] * 0.1
+                self.gromacs['func'] = 5
+            else:
+                raise NotImplementedError
+
+
+
+        elif isinstance(self, DihedralType):
+            if reqformat == 'gromacs' and self.format == 'charmm':
+                for dih in self.charmm['param']:
+                    convdih = {}
+                    convdih['kchi']  = dih['kchi'] * 4.184
+                    convdih['n']     = dih['n']
+                    convdih['delta'] = dih['delta']
+                    self.gromacs['param'].append(convdih)
+                self.gromacs['func'] = 9
+            else:
+                raise NotImplementedError
+
+
+
+        elif isinstance(self, ImproperType):
+            if reqformat == 'gromacs' and self.format == 'charmm':
+                for imp in self.charmm['param']:
+                    convimp = {}
+                    convimp['kpsi']  = imp['kpsi'] * 2 * 4.184
+                    convimp['psi0']  = imp['psi0']
+                    if imp.get('n', False):
+                        convimp['n']     = imp['n']
+                    self.gromacs['param'].append(convimp)
+                self.gromacs['func'] = 2
+
+                # self.gromacs['param']['kpsi'] = self.charmm['param']['kpsi'] * 2 * 4.184
+                # self.gromacs['param']['psi0'] = self.charmm['param']['psi0']
+                # self.gromacs['func'] = 2
+            else:
+                raise NotImplementedError
+
+
+
+        elif isinstance(self, CMapType):
+            if reqformat == 'gromacs' and self.format == 'charmm':
+                self.gromacs['param']= [n*4.184 for n in self.charmm['param']]
+                self.gromacs['func'] = 1
+            else:
+                raise NotImplementedError
+
+
+
+        elif isinstance(self, InteractionType):
+            if reqformat == 'gromacs' and self.format == 'charmm':
+                if self.charmm['param']['lje'] is not None:
+                    self.gromacs['param']['lje'] = abs(self.charmm['param']['lje']) * 4.184
+                    self.gromacs['param']['ljl'] = self.charmm['param']['ljl'] * 0.1 /  (2**(1.0/6.0))   # no *2
+                else:
+                    self.gromacs['param']['lje'] = None
+                    self.gromacs['param']['ljl'] = None
+
+                if self.charmm['param']['lje14'] is not None:
+                    self.gromacs['param']['lje14'] = abs(self.charmm['param']['lje14']) * 4.184
+                    self.gromacs['param']['ljl14'] = self.charmm['param']['ljl14'] * 0.1 / (2**(1.0/6.0))
+                else:
+                    self.gromacs['param']['lje14'] = None
+                    self.gromacs['param']['ljl14'] = None
+            else:
+                raise NotImplementedError
+
+        else:
+            raise NotImplementedError
+
```

Comment:
Since this is subclassed by a bunch of types, maybe defining a ``__repr__`` would help with debugging.